### ISSUE TITLE
"Add missing parameters to ABAPDoc" rules for unit-test methods

### ISSUE THREAD

**Issue by suynwa** [SAP/abap-cleaner#103]
I have a minor gripe regarding the rule _Add missing parameters to ABAPDoc_ in the context of aUnit methods. 

Before running abapCleaner:
```abap
    "! It should not add missing param CX_STATIC_CHECK 
    methods SHOULD_NOT_ADD_ABAP_DOC for testing raising CX_STATIC_CHECK.
```
After running abapCleaner:
```abap
    "! It should not add missing param CX_STATIC_CHECK
    "!
    "! @raising cx_static_check | 
    methods SHOULD_NOT_ADD_ABAP_DOC for testing raising CX_STATIC_CHECK.
```
IMO, this rule should not be run in the context of aUnit methods. Maybe it is a case of personal taste 😉

**Comment by jmgrassau** [comment]
Hi Suhas, 

just to be sure, the idea would be to add an option suppressing this rule in the context of FOR TESTING methods? (Or even FOR TESTING classes as a whole, so it wouldn't run on helper methods in those classes, either?) 

If this is not specific to test code versus product code, this option would help:

![image](https://github.com/SAP/abap-cleaner/assets/79639298/ac391322-9ee1-4db7-8c7c-255ef1b0decf)

Kind regards,
Jörg-Michael

**Comment by suynwa** [comment]
Hi Jörg-Michael,

I would rather skip the rule for aunit methods instead of providing an option. I don't think any one would like to document exceptions for aunit methods.

```abap
methods SHOULD_NOT_ADD_ABAP_DOC for testing raising CX_STATIC_CHECK.
```

In case some one wants to document these exceptions, (s)he can use the ADT quickfix (Ctrl+1) "Add missing parameters to documentation".

KR,
Suhas

**Comment by jmgrassau** [comment]
Hi Suhas, 

alright, this will be changed with the next release, meaning that no ABAP Doc line is added for exceptions of test methods. Nevertheless, if exceptions are already documented,
* existing ABAP Doc lines are kept if the exceptions are still raised
* they are removed if the exception was removed
* their order is aligned.

![image](https://github.com/SAP/abap-cleaner/assets/79639298/f62fc18c-99ae-4d0c-96cc-d7b2972a47c5)

Instead of an option, I added some documentation …

![image](https://github.com/SAP/abap-cleaner/assets/79639298/a64cf0f8-6339-42da-9307-90682b781d49)

… so if someone comes up with convincing arguments why they need this, we could still add an option for it at a later point in time.

Kind regards,
Jörg-Michael
